waddrmgr/db.go

```diff
@@ -292,6 +376,64 @@ func fetchWatchingOnly(tx walletdb.Tx) (bool, error) {
 	return buf[0] != 0, nil
 }
 
+// fetchNumAllAddresses loads the metadata - number of all addrs - from the database.
+func fetchNumAllAddresses(tx walletdb.Tx) (uint32, error) {
+	bucket := tx.RootBucket().Bucket(metaBucketName)
+
+	val := bucket.Get(numAllAddrsName)
+	// Return 0 if the metadata is uninitialized
+	if val == nil {
```

Comment:
Along the same lines as mentioned elsewhere, I would much rather this information be initialized at creation time with an error if it's missing here.  That is much more deterministic and doesn't hide errors as easily.